### PR TITLE
Config change set

### PR DESCRIPTION
**Pull request by madushadhanushka**
**Purpose**
Apply change config set configuration script before running test scenarios
**Note**
In testgrid.yaml file "config-change-sets" key should be rename as "configChangeSets" and "applies-to" should be rename as "appliesTo".

### CODE REVIEW

File: common/src/main/java/org/wso2/testgrid/common/ConfigChangeSet.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (harshanL) [inline common/src/main/java/org/wso2/testgrid/common/ConfigChangeSet.java]: License should be changed to 2018.

File: core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java [hunk 2]
```diff
@@ -136,23 +143,89 @@ private void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploy
                 }
             }
         }
+
+        Set<String> appliedSenarios = new HashSet<String>();
+        if (testPlan.getScenarioConfig().getConfigChangeSets() != null) {
```
Comment (harshanL) [inline core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java]: Please assign "testPlan.getScenarioConfig().getConfigChangeSets()" to a variable and use it.

File: core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java [hunk 3]
```diff
@@ -136,23 +143,89 @@ private void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploy
...
+            for (ConfigChangeSet configChangeSet : testPlan.getScenarioConfig().getConfigChangeSets()) {
+                for (String configScenario : configChangeSet.getAppliesTo()) {
+                    for (TestScenario testScenario : testPlan.getTestScenarios()) {
+                        if (configScenario.equals(testScenario.getName())) {
+                            appliedSenarios.add(configScenario);
```
Comment (harshanL) [inline core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java]: Variable "appliedSenarios" is misspelled.

File: core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java [hunk 4]
```diff
@@ -136,23 +143,89 @@ private void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploy
...
+                }
             }
-            try {
-                persistTestScenario(testScenario);
-            } catch (TestPlanExecutorException e) {
+        }
+    }
+
+    /**
+     * apply config change set script before run test scenarios
```
Comment (harshanL) [inline core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java]: Change "apply" to proper case.

File: core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java [hunk 5]
```diff
@@ -136,23 +143,89 @@ private void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploy
...
+                "config-sets", configChangeSet.getName());
+        ShellExecutor shellExecutor = new ShellExecutor(configChangeSetPath);
+        String bashCommand;
+        if (isInit) {
+            bashCommand = "bash apply-config.sh";
+        } else {
+            bashCommand = "bash revert-config.sh";
+        }
+        try {
+
```
Comment (pasindujw) [inline core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java]: We can remove the empty line.

File: common/src/main/java/org/wso2/testgrid/common/ConfigChangeSet.java (hunk 1 above)
Comment (madushadhanushka) [reply common/src/main/java/org/wso2/testgrid/common/ConfigChangeSet.java]: fixed

**Pull request merged**